### PR TITLE
Add command to set default value for service type and management type

### PR DESCRIPTION
**Pull request by eloaverona**
PR adds the commands bellow that will support creating circuits via cli
- `splinter circuit default`:
  - `set service-type <service-type>`: set a default service-type, the default value will be used during circuit creation if no service type is passed.
  - `set management-type <management-type>`: set a default management-type, the default value will be used during circuit creation if no management type is passed.
  - ` get service-type`: get the default value for service-type
  - `get management-type`: get the default value for management-type
  - `list`: list all defaults
  - `unset service-type` unset the default value for service-type
  - `unset management-type`: unset the default value for management-type

For more context on how these command can be used refer to [Shawn's comment](https://zenhub.cglcloud.com/app/workspaces/product-core-5d37298b0add7306b30efde2/issues/digitalsupplychain/product-core/1743#issuecomment-110397).

### CODE REVIEW

File: cli/src/store/default_value/yaml_store.rs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        if let Some(index) = existing_default_index {
+            defaults.remove(index);
+        }
+
+        defaults.push(SerdeDefaultValue::from(new_default_value));
+
+        self.save(&defaults)?;
+
+        Ok(())
+    }
```
Comment (agunde406) [inline cli/src/store/default_value/yaml_store.rs]: Can you add a blank line here

File: cli/src/main.rs [hunk 2]
```diff
@@ -319,6 +319,30 @@ fn run() -> Result<(), CliError> {
...
+                        .about("Manage default values for circuit creation")
+                        .subcommand(
+                            SubCommand::with_name("set")
+                                .about("Set a default value")
+                                .setting(AppSettings::SubcommandRequiredElseHelp)
+                                .subcommand(
+                                    SubCommand::with_name("service-type")
+                                        .about("Set default value for service type")
+                                        .arg(
+                                            Arg::with_name("service_type")
```
Comment (agunde406) [inline cli/src/main.rs]: This should have a `.value_name("service-type")`

File: cli/src/main.rs [hunk 3]
```diff
@@ -341,6 +341,23 @@ fn run() -> Result<(), CliError> {
                                                 .long("force")
                                                 .help("Overwrite default if it is already set"),
                                         ),
+                                )
+                                .subcommand(
+                                    SubCommand::with_name("management-type")
+                                        .about("Set default value for management type")
+                                        .arg(
+                                            Arg::with_name("management_type")
```
Comment (agunde406) [inline cli/src/main.rs]: Needs `.value_name("management-type")`

File: cli/src/action/circuit/defaults.rs [hunk 4]
```diff
@@ -51,6 +51,16 @@ impl Action for SetServiceTypeDefaultAction {
     }
 }
 
+pub struct UnsetServiceTypeDefaultAction;
```
Comment (agunde406) [inline cli/src/action/circuit/defaults.rs]: Typo in commit message: commant -> command

File: cli/src/main.rs [hunk 5]
```diff
@@ -373,6 +373,15 @@ fn run() -> Result<(), CliError> {
                                         .about("Unset default value for management type"),
                                 ),
                         )
+                        .subcommand(
+                            SubCommand::with_name("get")
```
Comment (agunde406) [inline cli/src/main.rs]: Any reason we are doing "get" here instead of "show"?

Comment (peterschwarz) [reply cli/src/main.rs]: It probably should be

File: cli/src/main.rs [hunk 6]
```diff
@@ -319,6 +319,30 @@ fn run() -> Result<(), CliError> {
...
+                )
+                .subcommand(
+                    SubCommand::with_name("default")
+                        .about("Manage default values for circuit creation")
+                        .subcommand(
+                            SubCommand::with_name("set")
+                                .about("Set a default value")
+                                .setting(AppSettings::SubcommandRequiredElseHelp)
+                                .subcommand(
+                                    SubCommand::with_name("service-type")
```
Comment (peterschwarz) [inline cli/src/main.rs]: Is there a reason why this is a separate sub command vs `set <value key> <value>` where `<value key>` is in the set `{service-type, management-type}`.  That seems more extensible, where we can add more default keys, as we decide which can be configured with defaults.

File: cli/src/store/default_value/yaml_store.rs (hunk 1 above)
Comment (eloaverona) [reply cli/src/store/default_value/yaml_store.rs]: Added

File: cli/src/main.rs (hunk 5 above)
Comment (eloaverona) [reply cli/src/main.rs]: I changed it to show. Also added a commit to change the `node alias get` to `node alias show`, which was in the PR merged yesterday

File: cli/src/main.rs (hunk 6 above)
Comment (eloaverona) [reply cli/src/main.rs]: Okay, I updated the PR to be as you suggested.

File: cli/src/action/circuit/defaults.rs [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+        let name = match args.value_of("name") {
+            Some(key) => key,
+            None => return Err(CliError::ActionError("name is required".into())),
+        };
+
+        let key = match name {
+            "service-type" => SERVICE_TYPE_KEY,
+            "management-type" => MANAGEMENT_TYPE_KEY,
+            _ => return Err(CliError::ActionError("Invalid default name".into())),
```
Comment (peterschwarz) [inline cli/src/action/circuit/defaults.rs]: This should probably be `format!("{} is not a valid default name", name)`

Comment (peterschwarz) [reply cli/src/action/circuit/defaults.rs]: Likewise  in the other places used.

Comment (eloaverona) [reply cli/src/action/circuit/defaults.rs]: updated

**Pull request merged**